flytekit/clients/raw.py

```diff
@@ -86,7 +87,11 @@ def handler(*args, **kwargs):
                         refresh_handler_fn = _get_refresh_handler(_creds_config.AUTH_MODE.get())
                         refresh_handler_fn(args[0])
                     else:
-                        raise
+                        if i == (retries - 1):
+                            raise
+                        else:
+                            _logging.error(f"Non-auth RPC error {e}, sleeping 5 seconds and retrying")
+                            time.sleep(5)
```

Comment:
Wait, we are sleeping for 5 seconds? 

That's extremely high for all calls. Can we do it only on waitforcompletion. ? This will kill all synchronous calls to Flyteadmin. Imagine a service using flytekit to call flyte. 5 seconds will kill the callers of the service

Reply:
So even a get status request can be invoked behind a service ? That way we can't do retries at all or maybe just 1-2

Reply:
Also do it only for Get calls

Reply:
if that's the case, maybe just scrap this PR entirely.  Maybe we can just do retries within the wait_for_completion function?  Where is that function defined?

Reply:
I am planning to create another handler here with retries to use for all get/idempotent calls only.. 

Also would retry with lower wait times (a few 100ms with exponential back off)

Reply:
done, PTAL